### ISSUE TITLE
Found the Trim disabled by starting the Mac in Monterey

### ISSUE THREAD

**Issue by OKonnel** [dortania/OpenCore-Legacy-Patcher#288]
Good evening everyone.
Maybe I'm talking nonsense about the disk's Trim; in this case it will be sufficient to close the post.
Premise:
in my iMac I have the 128GB Apple SSD disk that belonged to the Fusion Drive (on which I installed Monterey Beta) and the Crucial MX500 SSD 2TB disk that I put in place of the mechanical HD and (on which I installed Big Sur and all my applications and Data).
A little while ago I realized that while with OCLP and Big Sur I have always found both the Trim of the Apple 128 internal SSD and the Trim of the Crucial SSD active, instead, restarting the Mac from the Monterey disk, I saw that the Trim of the Crucial SSD internal (with Big Sur installed with FileVault) was disabled.
Until now, since I used OCLP to install Big Sur from scratch, I have never had to activate Trim myself, knowing full well that OCLP does.
Obviously, tonight I had to use the Terminal with the command
sudo trimforce enable
Tim is now active on both discs.
But shouldn't OCLP turn it on even when it comes to Monterey?
Thanks

**Comment by Ausdauersportler** [comment]
Before setting this one should figure out if and when it has advantages...

**Comment by lulujyc** [comment]
Seems like this issue can be restated as the thirdpartydrives setting in opencore broke. Might want to repeat test it and report to acidanthera if needed. Sorry I don't have 3rd party mac drives running Monterey right now but probably others can verify this.

**Comment by OKonnel** [comment]
I entered the problem because I think I read (perhaps in the OpenCore explanations?...), that EFI Opencore enables Trim by default.
Obviously I checked in System Information -> SATA and as I said before I found that Monterey has deactivated the Trim. After, OC EFI has not activated it anymore as always has. So I activated it from Terninale.
As to whether Trim is useful or not useful... there are those who say "yes" and those who say "no" o "perhaps" ;)
I think Trim serves to keep the SSDs that support it healthy and my Crucial MX500 supports Trim. I also read that it extends the life of SSDs and (maybe) that APFS works better.

**Comment by Ausdauersportler** [comment]
Hmm, my Samsung 860 EVO (not an Apple drive) has TRIM enabled (using OCLP 0.1.7 nightly build) running Monterey. This is a third party drive.

![TRIM](https://user-images.githubusercontent.com/71768134/121752012-933b0a00-cb0f-11eb-89bc-b5037c700346.png)

**Comment by OKonnel** [comment]
I started from a wrong assumption: I remembered wrongly ... since it's Micropatcher "install-setvars.sh" that "... includes disabling SIP, disabling authenticated root, and enabling TRIM on non-Apple SSDs." OpenCore, on the other hand, intervenes on Trim parameters only.
Thanks to insights suggested by @Ausdauersportler I found that APFS probably works better with TRIM disabled.
Finally - and close the thread - I found an [article](https://interface31.ru/tech_it/2015/04/mozhno-li-effektivno-ispolzovat-ssd-bez-podderzhki-trim.html) in Russian that I read with the Translator and it seems very comprehensive and complete.

Thank you all
